src/main/java/seedu/address/logic/parser/BatchDeleteCommandParser.java

```diff
@@ -18,17 +21,29 @@ public class BatchDeleteCommandParser implements Parser<BatchDeleteCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public BatchDeleteCommand parse(String args) throws ParseException {
-        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_DELETE_MONTH);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_DELETE_MONTH, PREFIX_COMPANY);
 
-        if (!argMultimap.getValue(PREFIX_DELETE_MONTH).isPresent()) {
+        boolean hasDeleteMonth = argMultimap.getValue(PREFIX_DELETE_MONTH).isPresent();
+        boolean hasCompany = argMultimap.getValue(PREFIX_COMPANY).isPresent();
+
+        if (hasDeleteMonth && hasCompany) {
+            throw new ParseException(String.format(MESSAGE_INVALID_TWO_FIELD, PREFIX_DELETE_MONTH, PREFIX_COMPANY));
+        } else if (!hasDeleteMonth && !hasCompany) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, BatchDeleteCommand.MESSAGE_USAGE));
         }
-        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_DELETE_MONTH);
 
-        DeleteMonth deleteMonth = ParserUtil.parseDeleteMonth(argMultimap.getValue(PREFIX_DELETE_MONTH).get());
+        if (hasDeleteMonth) {
+            argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_DELETE_MONTH);
+            DeleteMonth deleteMonth = ParserUtil.parseDeleteMonth(argMultimap.getValue(PREFIX_DELETE_MONTH).get());
+            assert !deleteMonth.equals(null) : "wrong input for deleteMonth";
+            return new BatchDeleteCommand(deleteMonth);
```

Comment:
So if the user inputs a prefix `dm/deletemonth` , they will not be allowed to indicate a company field as well, which is as intended, good job!